tests/stress_tests/memcheck_tests/tests.cpp

```diff
@@ -106,3 +106,51 @@ INSTANTIATE_TEST_CASE_P(MemCheckTests, MemCheckTestSuite,
                         ::testing::ValuesIn(
                                 generateTestsParams({"devices", "models"})),
                         getTestCaseName);
+
+TEST_P(MemCheckTestSuite, inference_with_streams) {
+    const auto nstreams = 2;
+    log_info("Inference of InferRequest from network: \"" << model
+                                                          << "\" with precision: \"" << precision
+                                                          << "\" for device: \"" << device << "\""
+                                                          << "\" with streams: \"" << nstreams);
+    if ((device != "CPU") && (device != "GPU"))
+        throw std::invalid_argument("This device is not supported");
+
+    auto test_pipeline = [&] {
+        MemCheckPipeline memCheckPipeline;
+        Core ie;
+        std::map<std::string, std::string> config;
+        const std::string key = device + "_THROUGHPUT_STREAMS";
+        InferRequest inferRequest;
+
+        config[key] = std::to_string(nstreams);
+        ie.GetVersions(device);
+        ie.SetConfig(config, device);
+
+        CNNNetwork cnnNetwork = ie.ReadNetwork(model);
+        ExecutableNetwork exeNetwork = ie.LoadNetwork(cnnNetwork, device);
+        auto batchSize = cnnNetwork.getBatchSize();
+        batchSize = batchSize != 0 ? batchSize : 1;
+        const ConstInputsDataMap inputsInfo(exeNetwork.GetInputsInfo());
+
+        for (int counter = 0; counter < nstreams; counter++) {
+            inferRequest = exeNetwork.CreateInferRequest();
```

Comment:
Why not `OPTIMAL_NUMBER_OF_INFER_REQUESTS`?

Reply:
Thanks for the idea! @akoryach please, take a look

Reply:
Implemented by using `OPTIMAL_NUMBER_OF_INFER_REQUESTS`

Reply:
@nkogteva could you also comment, should here be enabled `.StartAsync()` instead of consecutive inference?